extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaRecordSupplier.java

```diff
@@ -157,6 +158,14 @@ public Long getEarliestSequenceNumber(StreamPartition<Integer> partition)
     return nextPos;
   }
 
+  @Override
+  public boolean isOffsetAvailable(StreamPartition<Integer> partition, OrderedSequenceNumber<Long> offset)
```

Comment:
Does this needs a lock on recordSupplierLock?

Reply:
Maybe we can call getOffsetFromStreamForPartition in here instead?

Reply:
Yes, thanks!